pepper-apis/src/main/java/org/broadinstitute/ddp/db/ActivityInstanceDao.java

```diff
@@ -266,17 +266,9 @@ public Long getActivityInstanceIdByGuid(Handle handle, String guid) {
             String studyGuid,
             String isoLanguageCode
     ) {
-        JdbiFormTypeActivityInstanceStatusType jdbiFormTypeActivityInstanceStatusType =
-                handle.attach(JdbiFormTypeActivityInstanceStatusType.class);
-        Map<String, Blob> formTypeAndStatusTypeToIcon = new HashMap<>();
-        List<IconBlobDto> iconBlobs = jdbiFormTypeActivityInstanceStatusType.getIconBlobs(studyGuid);
-        // Transforming a List of icon blobs into a Map having form type and activity instance status type
-        // concatenated with "-" as a delimiter as keys and icon blobs as values, e.g. { "CONSENT-5": "<icon blob>" }
-        iconBlobs.forEach(blob -> formTypeAndStatusTypeToIcon.put(
-                blob.getFormType() + "-" + blob.getStatusTypeCode(), blob.getIconBlob()
-                )
-        );
-        Collection<ActivityInstanceSummary> activitySummaries = new ArrayList<>();
+        ActivityDefStore activityDefStore = ActivityDefStore.getInstance();
+        Map<String, Blob> formTypeAndStatusTypeToIcon = activityDefStore.findActivityStatusIcons(handle, studyGuid);
```

Comment:
Key insight here is to cache the icons since they rarely change (if at all) instead of querying for them every time we invoke this method. This saves several milliseconds.